sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/implementation/AppConfigurationPullRefresh.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.AsyncResult;
+import org.springframework.stereotype.Component;
+
+import com.azure.spring.cloud.config.AppConfigurationRefresh;
+import com.azure.spring.cloud.config.health.AppConfigurationStoreHealth;
+import com.azure.spring.cloud.config.implementation.AppConfigurationRefreshUtil.RefreshEventData;
+import com.azure.spring.cloud.config.pipline.policies.BaseAppConfigurationPolicy;
+import com.azure.spring.cloud.config.properties.AppConfigurationProviderProperties;
+
+/**
+ * Enables checking of Configuration updates.
+ */
+@Component
+public class AppConfigurationPullRefresh implements AppConfigurationRefresh {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AppConfigurationPullRefresh.class);
+
+    private final AtomicBoolean running = new AtomicBoolean(false);
+
+    private ApplicationEventPublisher publisher;
+
+    private final AppConfigurationProviderProperties appProperties;
+
+    private final AppConfigurationReplicaClientFactory clientFactory;
+
+    private final Duration refreshInterval;
+
+    /**
+     * Component used for checking for and triggering configuration refreshes.
+     *
+     * @param properties Client properties to check against.
+     * @param appProperties Library properties for configuring backoff
+     * @param clientFactory Clients stores used to connect to App Configuration.
+     */
+    public AppConfigurationPullRefresh(AppConfigurationProviderProperties appProperties,
+        AppConfigurationReplicaClientFactory clientFactory, Duration refreshInterval) {
+        this.appProperties = appProperties;
+        this.refreshInterval = refreshInterval;
+        this.clientFactory = clientFactory;
+
+    }
+
+    @Override
+    public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+        this.publisher = applicationEventPublisher;
+    }
+
+    /**
+     * Checks configurations to see if configurations should be reloaded. If the refresh interval has passed and a
+     * trigger has been updated configuration are reloaded.
+     *
+     * @return Future with a boolean of if a RefreshEvent was published. If refreshConfigurations is currently being run
+     * elsewhere this method will return right away as <b>false</b>.
+     */
+    @Async
+    public Future<Boolean> refreshConfigurations() {
+        return new AsyncResult<>(refreshStores());
+    }
+
+    /**
+     * Soft expires refresh interval. Sets amount of time to next refresh to be a random value between 0 and 15 seconds,
+     * unless value is less than the amount of time to the next refresh check.
+     * @param endpoint Config Store endpoint to expire refresh interval on.
+     * @param syncToken syncToken to verify the latest changes are available on pull
+     */
+    public void expireRefreshInterval(String endpoint, String syncToken) {
```

Comment:
It's weird to pass a syncToken to this `expireRefreshInterval` method. Why does it need a syncToken?

Reply:
This is for push refresh. A notification came to app config to refresh, but our design says it doesn't trigger a refresh only expires the refresh period. The Sync Token came from the request so we know that when we do refresh we are synced.